org.lflang/src/org/lflang/util/FileUtil.java

```diff
@@ -104,6 +104,59 @@ public static IPath toIPath(URI uri) throws IOException {
     public static String toUnixString(Path path) {
         return path.toString().replace('\\', '/');
     }
+    
+    /**
+     * Parse the string as file location and return it as URI.
+     * Supports URIs, plain file paths, and paths relative to a model.
+     * 
+     * @param path the file location as string.
+     * @param resource the model resource this file should be resolved relatively. May be null.
+     * @return the (Java) URI or null if no file can be located.
+     */
+    public static java.net.URI locateFile(String path, Resource resource) {
+        // Check if path is URL
+        try {
+            var uri = new java.net.URI(path);
+            if(uri.getScheme() != null) { // check if path was meant to be a URI
+                return uri;
+            }
+        } catch (Exception e) {
+            // nothing
+        }
+        // Check if path exists as it is
+        File file = new File(path);
+        if (file.exists()) {
+            try {
+                return file.toURI();
+            } catch (Exception e) {
+                // nothing
+            }
+        }
+        // Check if path is relative to LF file
+        if (resource != null) {
```

Comment:
Since the introduction of `FileConfig`, we have been interpreting paths relative to the project root, which is understood to be the parent of the `src` directory. We should probably include this possibility in the search. Would it make sense to pass in a `FileConfig` instance? We could also take the resource and find the `src` in its path, but that would be duplicating logic we already have in `FileConfig`...

Reply:
I moved this code from the diagram package to that utility class in order to use the same code base for the validator, however, in the diagram synthesis and the validator, there is only the model. How do I obtain the a `FileConfig` for a model in that context? I looks like `FileConfig` is only available in the generator.

Reply:
You're right. I think this solution is fine, but it should also search relative to the root of the project (not just relative to the file itself). The root of the project is just the parent of the nearest containing `src` directory.

Reply:
Ah, yes, I can do that.